app/controllers/drops_controller.rb

```diff
@@ -69,8 +69,8 @@ def upvote
   private
 
     def set_drop
-      @drop = Drop.find(params[:id].to_i)
-    end
+      page_not_found unless @drop = Drop.find_by(id: params[:id].to_i)
```

Comment:
@carpodaster: Thanks for this explanation. It was really great, but if we want to show the error with a pretty nice form with cats and all that, we would need this method right?